### PR TITLE
Improve !info performance

### PR DESCRIPTION
**Pull request by Scott-Caldwell**
Really hoping to make this command more efficient...

Parallelizes API calls, adds caching in various places.

### CODE REVIEW

File: Modix.Services/Core/ChannelService.cs [hunk 1]
```diff
@@ -29,33 +41,62 @@ public class ChannelService : IChannelService
...
+                    ChannelId = channel.Id,
+                    GuildId = channel.GuildId,
+                    Name = channel.Name
+                }, cancellationToken);
+            }
+
+            transaction.Commit();
+        }
+
+        public async ValueTask<ITextChannel> GetChannelAsync(ulong guildId, ulong channelId)
```
Comment (JakenVeina) [inline Modix.Services/Core/ChannelService.cs]: A) The Discord.NET Socket client already caches ALL guild data in memory, except for messages.
B) I'm very much not comfortable caching objects for which we do not control the lifetime.

Comment (Scott-Caldwell) [reply Modix.Services/Core/ChannelService.cs]: That's fair, I wasn't sure how much D.Net cached

Comment (JakenVeina) [reply Modix.Services/Core/ChannelService.cs]: Have a look at the API of the `Socket` classes, you can usually tell. `SocketGuild`, for example, has a property` public IReadOnlyCollection<SocketGuildChannel> Channels { get; }` and a method `public SocketGuildChannel GetChannel(ulong id);`

**Comment by JakenVeina** [comment]
Last chance to re-write history? You could just force-push from before you added the caching.

**Comment by Scott-Caldwell** [comment]
I thought about that, but then I noticed I added a couple indexes as well so I decided not to revert the entire commit. Probably a candidate for a squash anyway

**Pull request merged**